### PR TITLE
Fix: nested collection lazy only includes the first element in the collection

### PR DESCRIPTION
**Pull request by Tofandel**
Might not be the exact same issue as #677 but definitely related

If you have a collection within a collection, including lazy keys will only include the key in the first item of the collection

### CODE REVIEW

File: tests/PartialsTest.php [hunk 1]
```diff
@@ -225,6 +225,37 @@
...
+        ]
+    ], DataCollection::class);
+
+    expect($data->toArray())->toMatchArray([
+        ['nested' => [[]]],
+        ['nested' => [[]]],
+        ['nested' => [[]]]
+    ]);
+
+    expect(dump($data->include('nested.{artist,year}')->toArray()))
```
Comment (Tofandel) [inline tests/PartialsTest.php:251]: Current output

```
array:3 [
  0 => array:1 [
    "nested" => array:1 [
      0 => array:2 [
        "artist" => "Rick Astley"
        "year" => 1987
      ]
    ]
  ]
  1 => array:1 [
    "nested" => array:1 [
      0 => []
    ]
  ]
  2 => array:1 [
    "nested" => array:1 [
      0 => []
    ]
  ]
] 
```

File: tests/PartialsTest.php [hunk 2]
```diff
@@ -225,6 +225,37 @@
...
+        ]
+    ], DataCollection::class);
+
+    expect($data->toArray())->toMatchArray([
+        ['nested' => [[]]],
+        ['nested' => [[]]],
+        ['nested' => [[]]]
+    ]);
+
+    expect($data->include('nested.{artist,year}')->toArray())
```
Comment (Tofandel) [inline tests/PartialsTest.php:251]: Current ouput

```
array:3 [
  0 => array:1 [
    "nested" => array:1 [
      0 => array:2 [
        "artist" => "Rick Astley"
        "year" => 1987
      ]
    ]
  ]
  1 => array:1 [
    "nested" => array:1 [
      0 => []
    ]
  ]
  2 => array:1 [
    "nested" => array:1 [
      0 => []
    ]
  ]
] 
```

File: src/Resolvers/VisibleDataFieldsResolver.php [hunk 3]
```diff
@@ -214,6 +214,7 @@ protected function resolveIncludedFields(
                 continue;
             }
 
+            $includedPartial = clone $includedPartial;
```
Comment (Tofandel) [inline src/Resolvers/VisibleDataFieldsResolver.php:217]: This is what I found to fix the issue, the problem is that it's looping over the elements of the collection but all the elements of the collection get the same instance of `includedPartial` which is mutated with `includedPartial->next()` and so `$includedPartial->getNested()` returns null on the second item

File: src/Resolvers/VisibleDataFieldsResolver.php [hunk 4]
```diff
@@ -230,13 +230,13 @@ protected function resolveIncludedFields(
             }
 
             if ($nested = $includedPartial->getNested()) {
+                $includedPartial = clone $includedPartial;
```
Comment (Tofandel) [inline src/Resolvers/VisibleDataFieldsResolver.php:233]:
```
This is what I found to fix the issue, the problem is that it's looping over the elements of the collection but all the elements of the collection get the same instance of includedPartial which is mutated with includedPartial->next() and so $includedPartial->getNested() returns null on the second item

```

**Comment by Tofandel** [comment]
I'm also noting that `currentPath` in `creationContext` is also affected by a similar issue
image

**Comment by rubenvanassche** [comment]
Hi @Tofandel thanks for the PR, the issue should be solved with the next version of data. I've implemented the system a bit differently.

**Comment by Tofandel** [comment]
Did you use the tests in this PR and did you also fix creationContext? Can you link the commits so I can check?

**Comment by rubenvanassche** [comment]
Some tests were not running that's been fixed now, when I paste your tests in the current data test suite they are green. As for creation context, could you send in a PR showing what's going on?

**Comment by Tofandel** [comment]
For creation context it's just this commit https://github.com/spatie/laravel-data/pull/720/commits/f0a1a2e7870d4c7a1e595074a477ab2d1ae461d5

Basically you are going into a nested state without calling prev like in the other method where it's used, it's a small quick fix

**Pull request closed without merging**